console/console-init/ui/src/modules/iot-device/components/CreateMetadata/MetaDataRow.tsx

```diff
@@ -29,16 +29,43 @@ export const MetaDataRow: React.FC<IMetaDataRow> = ({
   rowIndex
 }) => {
   const metadataRow = metadataList[rowIndex];
-  // console.log("metadataList", metadataList, "type", metadataRow.type);
+  const [validationStatus, setValidationStatus] = useState<"default" | "error">(
+    "default"
+  );
 
   const onSelectType = (typeValue: string) => {
+    const validationStatus = getValidationStatus(typeValue, metadataRow.value);
+    setValidationStatus(validationStatus);
     let updatedTypeMetadata = [...metadataList];
     updatedTypeMetadata[rowIndex].type = typeValue;
     if (isObjectOrArray(typeValue as any))
       updatedTypeMetadata[rowIndex].value = "";
     setMetadataList(updatedTypeMetadata);
   };
 
+  const getValidationStatus = (type: string, value: string) => {
+    let validationStatus: "default" | "error" = "default";
+    switch (type) {
+      case "string":
+        validationStatus = typeof value === "string" ? "default" : "error";
+        break;
+      case "number":
+        validationStatus = !isNaN(Number(value)) ? "default" : "error";
+        break;
+      case "boolean":
+        validationStatus =
+          value.toLowerCase() === "true" || value.toLowerCase() === "false"
+            ? "default"
+            : "error";
+        break;
+      case "datetime":
```

Comment:
not sure, this `datetime` type check will work or not. pleas re-check for datetime.